2 tests fail on aarch64 and ppc64le with `4.14.0`

Tests fail on aarch64 and ppc64le with 4.14.0: `test_regproj[sherpa.plot.dummy_backend] ` and `test_fit_single[Chi2XspecVar-False-False-2.1192140563]`
It seems to be only precision issues.

Log:
```
[  468s] =================================== FAILURES ===================================
[  468s] ___________________ test_regproj[sherpa.plot.dummy_backend] ____________________
[  468s] 
[  468s] old_numpy_printing = None, override_plot_backend = None
[  468s] 
[  468s]     def test_regproj(old_numpy_printing, override_plot_backend):
[  468s]         p = plot.RegionProjection()
[  468s]         r = p._repr_html_()
[  468s]     
[  468s]         check_empty(r, 'RegionProjection', nsummary=13)
[  468s]     
[  468s]         x = np.arange(5, 8, 0.5)
[  468s]         y = np.asarray([2, 3, 4, 5, 4, 3])
[  468s]         dy = y / 2
[  468s]         d = Data1D('n n', x, y, staterror=dy)
[  468s]     
[  468s]         m = Polynom1D()
[  468s]         m.c1.thaw()
[  468s]     
[  468s]         fit = Fit(d, m, stat=Chi2())
[  468s]         fr = fit.fit()
[  468s]         assert fr.succeeded
[  468s]     
[  468s]         p.prepare(min=(-2, -1), max=(2, 2), nloop=(10, 20))
[  468s]         p.calc(fit, m.c0, m.c1)
[  468s]     
[  468s]         r = p._repr_html_()
[  468s]         assert r is not None
[  468s]     
[  468s]         if plot.backend.name == 'pylab':
[  468s]             assert '<summary>RegionProjection</summary>' in r
[  468s]             assert '<svg ' in r
[  468s]             return
[  468s]     
[  468s]         print(r)
[  468s]         assert '<summary>RegionProjection (13)</summary>' in r
[  468s]     
[  468s] >       assert '<div class="dataname">parval0</div><div class="dataval">-0.5315772076542427</div>' in r
[  468s] E       assert '<div class="dataname">parval0</div><div class="dataval">-0.5315772076542427</div>' in '<style>/*\nCopyright (C) 2020  Smithsonian Astrophysical Observatory\n\n\n This program is free software; you can red... class="dataval">4</div><div class="dataname">log</div><div class="dataval">(False, False)</div></div></details></div>'
[  468s] 
[  468s] ../../BUILDROOT/python-sherpa-4.14.0-27.1.aarch64/usr/lib64/python3.9/site-packages/sherpa/plot/tests/test_plot_notebook.py:324: AssertionError
[  468s] ----------------------------- Captured stdout call -----------------------------
[  468s] <style>/*
[  468s] Copyright (C) 2020  Smithsonian Astrophysical Observatory
[  468s] 
[  468s] 
[  468s]  This program is free software; you can redistribute it and/or modify
[  468s]  it under the terms of the GNU General Public License as published by
[  468s]  the Free Software Foundation; either version 3 of the License, or
[  468s]  (at your option) any later version.
[  468s] 
[  468s]  This program is distributed in the hope that it will be useful,
[  468s]  but WITHOUT ANY WARRANTY; without even the implied warranty of
[  468s]  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
[  468s]  GNU General Public License for more details.
[  468s] 
[  468s]  You should have received a copy of the GNU General Public License along
[  468s]  with this program; if not, write to the Free Software Foundation, Inc.,
[  468s]  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
[  468s] 
[  468s] */
[  468s] 
[  468s] :root {
[  468s]   --sherpa-border-color: var(--jp-border-color2, #e0e0e0);
[  468s]   --sherpa-background-color: var(--jp-layout-color0, white);
[  468s]   --sherpa-background-color-row-even: var(--jp-layout-color1, white);
[  468s]   --sherpa-background-color-row-odd: var(--jp-layout-color2, #eeeeee);
[  468s] 
[  468s]   /* https://medium.com/ge-design/iot-cool-gray-is-a-great-background-color-for-data-visualization-ebf18c318418 */
[  468s]   --sherpa-background-color-dark1: #EBEFF2;
[  468s]   --sherpa-background-color-dark2: #D8E0E5;
[  468s] }
[  468s] 
[  468s] div.sherpa-text-fallback {
[  468s]     display: none;
[  468s] }
[  468s] 
[  468s] div.sherpa {
[  468s]     display: block;
[  468s] }
[  468s] 
[  468s] div.sherpa details summary {
[  468s]     display: list-item;  /* needed for notebook, not lab */
[  468s]     font-size: larger;
[  468s] }
[  468s] 
[  468s] div.sherpa details div.datavals {
[  468s]     display: grid;
[  468s]     grid-template-columns: 1fr 3fr;
[  468s]     column-gap: 0.5em;
[  468s] }
[  468s] 
[  468s] div.sherpa div.dataname {
[  468s]     font-weight: bold;
[  468s]     border-right: 1px solid var(--sherpa-border-color);
[  468s] }
[  468s] 
[  468s] div.sherpa div.dataval { }
[  468s] 
[  468s] div.sherpa div.datavals div:nth-child(4n + 1) ,
[  468s] div.sherpa div.datavals div:nth-child(4n + 2) {
[  468s]     background: var(--sherpa-background-color-row-odd);
[  468s] }
[  468s] 
[  468s] div.sherpa table.model tbody {
[  468s]     border-bottom: 1px solid var(--sherpa-border-color);
[  468s] }
[  468s] 
[  468s] div.sherpa table.model tr.block {
[  468s]     border-top: 1px solid var(--sherpa-border-color);
[  468s] }
[  468s] 
[  468s] div.sherpa table.model th.model-odd ,
[  468s] div.sherpa table.model th.model-even {
[  468s]     border-right: 1px solid var(--sherpa-border-color);
[  468s] }
[  468s] 
[  468s] div.sherpa table.model th.model-odd {
[  468s]     background: var(--sherpa-background-color-dark1);
[  468s] }
[  468s] 
[  468s] div.sherpa table.model th.model-even {
[  468s]     background: var(--sherpa-background-color-dark2);
[  468s] }
[  468s] 
[  468s] div.sherpa .failed {
[  468s]     background: orange;
[  468s]     font-size: large;
[  468s]     padding: 1em;
[  468s] }
[  468s] </style><div class="sherpa-text-fallback">&lt;sherpa.plot.RegionProjection object at 0xffffa79c8a30&gt;</div><div hidden class="sherpa"><details open><summary>RegionProjection (13)</summary><div class="datavals"><div class="dataname">parval0</div><div class="dataval">-0.5315772076542431</div><div class="dataname">parval1</div><div class="dataval">0.5854611101216837</div><div class="dataname">sigma</div><div class="dataval">(1, 2, 3)</div><div class="dataname">x0</div><div class="dataval">[-2.       -1.555556 -1.111111 -0.666667 -0.222222  0.222222  0.666667  1.111111
[  468s]   1.555556  2.       -2.       -1.555556 -1.111111 -0.666667 -0.222222  0.222222
[  468s]   0.666667  1.111111  1.555556  2.       -2.       -1.555556 -1.111111 -0.666667
[  468s]  -0.222222  0.222222  0.666667  1.111111  1.555556  2.       -2.       -1.555556
[  468s]  -1.111111 -0.666667 -0.222222  0.222222  0.666667  1.111111  1.555556  2.
[  468s]  -2.       -1.555556 -1.111111 -0.666667 -0.222222  0.222222  0.666667  1.111111
[  468s]   1.555556  2.       -2.       -1.555556 -1.111111 -0.666667 -0.222222  0.222222
[  468s]   0.666667  1.111111  1.555556  2.       -2.       -1.555556 -1.111111 -0.666667
[  468s]  -0.222222  0.222222  0.666667  1.111111  1.555556  2.       -2.       -1.555556
[  468s]  -1.111111 -0.666667 -0.222222  0.222222  0.666667  1.111111  1.555556  2.
[  468s]  -2.       -1.555556 -1.111111 -0.666667 -0.222222  0.222222  0.666667  1.111111
[  468s]   1.555556  2.       -2.       -1.555556 -1.111111 -0.666667 -0.222222  0.222222
[  468s]   0.666667  1.111111  1.555556  2.       -2.       -1.555556 -1.111111 -0.666667
[  468s]  -0.222222  0.222222  0.666667  1.111111  1.555556  2.       -2.       -1.555556
[  468s]  -1.111111 -0.666667 -0.222222  0.222222  0.666667  1.111111  1.555556  2.
[  468s]  -2.       -1.555556 -1.111111 -0.666667 -0.222222  0.222222  0.666667  1.111111
[  468s]   1.555556  2.       -2.       -1.555556 -1.111111 -0.666667 -0.222222  0.222222
[  468s]   0.666667  1.111111  1.555556  2.       -2.       -1.555556 -1.111111 -0.666667
[  468s]  -0.222222  0.222222  0.666667  1.111111  1.555556  2.       -2.       -1.555556
[  468s]  -1.111111 -0.666667 -0.222222  0.222222  0.666667  1.111111  1.555556  2.
[  468s]  -2.       -1.555556 -1.111111 -0.666667 -0.222222  0.222222  0.666667  1.111111
[  468s]   1.555556  2.       -2.       -1.555556 -1.111111 -0.666667 -0.222222  0.222222
[  468s]   0.666667  1.111111  1.555556  2.       -2.       -1.555556 -1.111111 -0.666667
[  468s]  -0.222222  0.222222  0.666667  1.111111  1.555556  2.       -2.       -1.555556
[  468s]  -1.111111 -0.666667 -0.222222  0.222222  0.666667  1.111111  1.555556  2.      ]</div><div class="dataname">x1</div><div class="dataval">[-1.       -1.       -1.       -1.       -1.       -1.       -1.       -1.
[  468s]  -1.       -1.       -0.842105 -0.842105 -0.842105 -0.842105 -0.842105 -0.842105
[  468s]  -0.842105 -0.842105 -0.842105 -0.842105 -0.684211 -0.684211 -0.684211 -0.684211
[  468s]  -0.684211 -0.684211 -0.684211 -0.684211 -0.684211 -0.684211 -0.526316 -0.526316
[  468s]  -0.526316 -0.526316 -0.526316 -0.526316 -0.526316 -0.526316 -0.526316 -0.526316
[  468s]  -0.368421 -0.368421 -0.368421 -0.368421 -0.368421 -0.368421 -0.368421 -0.368421
[  468s]  -0.368421 -0.368421 -0.210526 -0.210526 -0.210526 -0.210526 -0.210526 -0.210526
[  468s]  -0.210526 -0.210526 -0.210526 -0.210526 -0.052632 -0.052632 -0.052632 -0.052632
[  468s]  -0.052632 -0.052632 -0.052632 -0.052632 -0.052632 -0.052632  0.105263  0.105263
[  468s]   0.105263  0.105263  0.105263  0.105263  0.105263  0.105263  0.105263  0.105263
[  468s]   0.263158  0.263158  0.263158  0.263158  0.263158  0.263158  0.263158  0.263158
[  468s]   0.263158  0.263158  0.421053  0.421053  0.421053  0.421053  0.421053  0.421053
[  468s]   0.421053  0.421053  0.421053  0.421053  0.578947  0.578947  0.578947  0.578947
[  468s]   0.578947  0.578947  0.578947  0.578947  0.578947  0.578947  0.736842  0.736842
[  468s]   0.736842  0.736842  0.736842  0.736842  0.736842  0.736842  0.736842  0.736842
[  468s]   0.894737  0.894737  0.894737  0.894737  0.894737  0.894737  0.894737  0.894737
[  468s]   0.894737  0.894737  1.052632  1.052632  1.052632  1.052632  1.052632  1.052632
[  468s]   1.052632  1.052632  1.052632  1.052632  1.210526  1.210526  1.210526  1.210526
[  468s]   1.210526  1.210526  1.210526  1.210526  1.210526  1.210526  1.368421  1.368421
[  468s]   1.368421  1.368421  1.368421  1.368421  1.368421  1.368421  1.368421  1.368421
[  468s]   1.526316  1.526316  1.526316  1.526316  1.526316  1.526316  1.526316  1.526316
[  468s]   1.526316  1.526316  1.684211  1.684211  1.684211  1.684211  1.684211  1.684211
[  468s]   1.684211  1.684211  1.684211  1.684211  1.842105  1.842105  1.842105  1.842105
[  468s]   1.842105  1.842105  1.842105  1.842105  1.842105  1.842105  2.        2.
[  468s]   2.        2.        2.        2.        2.        2.        2.        2.      ]</div><div class="dataname">y</div><div class="dataval">[ 306.854444  282.795953  259.744431  237.699877  216.662291  196.631674
[  468s]   177.608025  159.591344  142.581632  126.578889  256.358683  234.414967
[  468s]   213.478221  193.548442  174.625632  156.709791  139.800918  123.899013
[  468s]   109.004077   95.11611   210.422976  190.594037  171.772066  153.957063
[  468s]   137.149029  121.347964  106.553866   92.766738   79.986577   68.213386
[  468s]   169.047325  151.333162  134.625967  118.92574   104.232482   90.546192
[  468s]    77.86687    66.194518   55.529133   45.870717  132.23173   116.632342
[  468s]   102.039923   88.454472   75.875989   64.304475   53.73993    44.182353
[  468s]    35.631744   28.088104   99.97619    86.491578   74.013934   62.543259
[  468s]    52.079553   42.622814   34.173045   26.730243   20.294411   14.865546
[  468s]    72.280705   60.910869   50.548001   41.192102   32.843171   25.501209
[  468s]    19.166215   13.838189    9.517133    6.203044   49.145275   39.890215
[  468s]    31.642123   24.401      18.166845   12.939659    8.719441    5.506191
[  468s]     3.29991     2.100597   30.569902   23.429617   17.296301   12.169953
[  468s]     8.050574    4.938164    2.832722    1.734248    1.642742    2.558206
[  468s]    16.554583   11.529074    7.510534    4.498962    2.494359    1.496724
[  468s]     1.506058    2.52236     4.545631    7.575869    7.09932     4.188587
[  468s]     2.284823    1.388027    1.498199    2.61534     4.73945     7.870528
[  468s]    12.008574   17.153589    2.204112    1.408155    1.619167    2.837146
[  468s]     5.062095    8.294012   12.532897   17.778751   24.031573   31.291363
[  468s]     1.86896     3.187779    5.513566    8.846322   13.186046   18.532738
[  468s]    24.8864     32.247029   40.614627   49.989194    6.093863    9.527457
[  468s]    13.968021   19.415552   25.870052   33.331521   41.799958   51.275363
[  468s]    61.757737   73.247079   14.878821   20.427192   26.982531   34.544838
[  468s]    43.114114   52.690358   63.273571   74.863752   87.460902  101.06502
[  468s]    28.223835   35.886981   44.557096   54.234179   64.918231   76.609251
[  468s]    89.30724   103.012197  117.724122  133.443016   46.128904   55.906826
[  468s]    66.691717   78.483576   91.282404  105.0882    119.900964  135.720697
[  468s]   152.547398  170.381068   68.594029   80.486727   93.386393  107.293028
[  468s]   122.206632  138.127203  155.054744  172.989252  191.93073   211.879175
[  468s]    95.619209  109.626683  124.641125  140.662536  157.690915  175.726263
[  468s]   194.768579  214.817863  235.874116  257.937338  127.204444  143.326694
[  468s]   160.455912  178.592099  197.735254  217.885377  239.042469  261.206529
[  468s]   284.377558  308.555556]</div><div class="dataname">levels</div><div class="dataval">[  3.606863   7.491188  13.140272]</div><div class="dataname">min</div><div class="dataval">[-2, -1]</div><div class="dataname">max</div><div class="dataval">[2, 2]</div><div class="dataname">nloop</div><div class="dataval">(10, 20)</div><div class="dataname">delv</div><div class="dataval">None</div><div class="dataname">fac</div><div class="dataval">4</div><div class="dataname">log</div><div class="dataval">(False, False)</div></div></details></div>
[  468s] ____________ test_fit_single[Chi2XspecVar-False-False-2.1192140563] ____________
[  468s] 
[  468s] stat = <class 'sherpa.stats.Chi2XspecVar'>, usestat = False, usesys = False
[  468s] finalstat = 2.1192140563
[  468s] 
[  468s]     @pytest.mark.parametrize("stat,usestat,usesys,finalstat", [
[  468s]         (LeastSq, False, False, fit_lsq),
[  468s]         (LeastSq, True, True, fit_lsq),
[  468s]         (Chi2, True, True, fit_chi2_tt),
[  468s]         (Chi2, True, False, fit_chi2_tf),
[  468s]         (Chi2Gehrels, True, True, fit_chi2_tt),
[  468s]         (Chi2Gehrels, True, False, fit_chi2_tf),
[  468s]         (Chi2Gehrels, False, False, fit_gehrels_ff),
[  468s]         (Chi2Gehrels, False, True, fit_gehrels_ft),
[  468s]         (Chi2DataVar, True, True, fit_chi2_tt),
[  468s]         (Chi2DataVar, True, False, fit_chi2_tf),
[  468s]         (Chi2XspecVar, True, True, fit_chi2_tt),
[  468s]         (Chi2XspecVar, True, False, fit_chi2_tf),
[  468s]         (Chi2XspecVar, False, False, fit_xvar_ff),
[  468s]         (Chi2XspecVar, False, True, fit_xvar_ft),
[  468s]         (Chi2ConstVar, True, True, fit_chi2_tt),
[  468s]         (Chi2ConstVar, True, False, fit_chi2_tf),
[  468s]         (Chi2ConstVar, False, False, fit_cvar_ff),
[  468s]         (Chi2ConstVar, False, True, fit_cvar_ft),
[  468s]         (Chi2ModVar, True, True, fit_mvar_xt),
[  468s]         (Chi2ModVar, True, False, fit_mvar_xf),
[  468s]         (Chi2ModVar, False, False, fit_mvar_xf),
[  468s]         (Chi2ModVar, False, True, fit_mvar_xt),
[  468s]     
[  468s]         (Cash, True, True, fit_cash),
[  468s]         (Cash, True, False, fit_cash),
[  468s]         (Cash, False, False, fit_cash),
[  468s]         (Cash, False, True, fit_cash),
[  468s]     
[  468s]         (CStat, True, True, fit_cstat),
[  468s]         (CStat, True, False, fit_cstat),
[  468s]         (CStat, False, False, fit_cstat),
[  468s]         (CStat, False, True, fit_cstat),
[  468s]     ])
[  468s]     def test_fit_single(stat, usestat, usesys, finalstat):
[  468s]         """Check that the fit method works: single dataset, successful fit
[  468s]     
[  468s]         This is a minimal test, in that it just checks the
[  468s]         final statistic of the fit (not the model parameters).
[  468s]         The data and models are not designed to give good fit
[  468s]         results.
[  468s]         """
[  468s]     
[  468s]         statobj = stat()
[  468s]         fit = setup_stat_single(statobj, usestat, usesys)
[  468s]         assert fit.method.name == 'levmar'
[  468s]         fr = fit.fit()
[  468s]         assert fr.succeeded
[  468s] >       assert_almost_equal(fr.statval, finalstat)
[  468s] E       AssertionError: 
[  468s] E       Arrays are not almost equal to 7 decimals
[  468s] E        ACTUAL: 2.119359185687012
[  468s] E        DESIRED: 2.1192140563
[  468s] 
[  468s] ../../BUILDROOT/python-sherpa-4.14.0-27.1.aarch64/usr/lib64/python3.9/site-packages/sherpa/tests/test_fit_unit.py:1882: AssertionError
```

--- a/.github/workflows/ci-pip-arch.yml
+++ b/.github/workflows/ci-pip-arch.yml
@@ -1,21 +1,27 @@
-name: Pip CI (arch)
+name: Weekly cron (arch)
 
-# AstroPy does this weekly; for now I need to get this working first
-#
-on: [push, pull_request]
+on:
+  schedule:
+    # when is a good time?
+    - cron: '0 23 * * SUN'
 
 jobs:
   tests:
     name: ${{ matrix.arch }}
     runs-on: ubuntu-18.04
+    if: (github.repository == 'sherpa/sherpa' && github.event_name == 'schedule')
 
     strategy:
+      fail-fast: false
       matrix:
         include:
           - arch: aarch64
 
           # The tests on ppc64le fail rather spectacularly,
-          # and it's not obvious why, so skip for now.
+          # and it's not obvious why, so skip for now. AstroPy also
+          # see some interesting errors so maybe there are problems
+          # with the emulation or NumPy or ...
+          # https://github.com/astropy/astropy/pull/11697
           #
           # - arch: ppc64le
 
@@ -65,6 +71,9 @@ jobs:
           source tests/bin/activate
 
           # without this we get configure errors when building fftw and I don't know why
+          # (this adds the --disable-dependency-tracking option). A google suggests it
+          # could be a make vs gmake issue but we can't easily change this (and it may
+          # not be the actual problem).
           #
           sed -i.orig "s|#configure=None|configure=--disable-maintainer-mode --enable-stuberrorlib --disable-shared --enable-shared=libgrp,stklib --disable-dependency-tracking|" setup.cfg
 

--- a/.github/workflows/ci-pip-arch.yml
+++ b/.github/workflows/ci-pip-arch.yml
@@ -4,12 +4,15 @@ on:
   schedule:
     # when is a good time?
     - cron: '0 23 * * SUN'
+  pull_request:
+    # We also want this workflow triggered if the 'CI arch' label is added
+    # or present when PR is updated
+    types: [synchronize, labeled]
 
 jobs:
   tests:
-    name: ${{ matrix.arch }}
     runs-on: ubuntu-18.04
-    if: (github.repository == 'sherpa/sherpa' && github.event_name == 'schedule')
+    if: (github.repository == 'sherpa/sherpa' && (github.event_name == 'schedule' || contains(github.event.pull_request.labels.*.name, 'CI arch')))
 
     strategy:
       fail-fast: false

--- a/.github/workflows/ci-pip-arch.yml
+++ b/.github/workflows/ci-pip-arch.yml
@@ -1,0 +1,74 @@
+name: Pip CI (arch)
+
+# AstroPy does this weekly; for now I need to get this working first
+#
+on: [push, pull_request]
+
+jobs:
+  tests:
+    name: ${{ matrix.arch }}
+    runs-on: ubuntu-18.04
+
+    strategy:
+      matrix:
+        include:
+          - arch: aarch64
+
+          # The tests on ppc64le fail rather spectacularly,
+          # and it's not obvious why, so skip for now.
+          #
+          # - arch: ppc64le
+
+          # We currently do not have any user requests for this
+          # architecture.
+          #
+          # - arch: s390x
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v2
+      # As we do not include an I/O backend it is not worth checking out the data submodule.
+      #
+      # with:
+      #   submodules: 'True'
+
+    - name: Build and test
+      uses: uraimo/run-on-arch-action@v2.1.1
+      with:
+        arch: ${{ matrix.arch }}
+        distro: bullseye
+
+        shell: /bin/bash
+
+        # Based on AstroPy
+        install: |
+          echo "deb http://deb.debian.org/debian bullseye-backports main" >> /etc/apt/sources.list
+          apt-get update -q -y
+          apt-get install -q -y git \
+                                g++ \
+                                gfortran \
+                                flex \
+                                bison \
+                                make \
+                                pkg-config \
+                                python3 \
+                                python3-dev \
+                                python3-configobj \
+                                python3-numpy \
+                                python3-venv
+
+        # Follow AstroPy for now and do not install any external packages
+        # (I/O, plotting, or DS9).
+        #
+        run: |
+          python3 -m venv --system-site-packages tests
+          source tests/bin/activate
+
+          # without this we get configure errors when building fftw and I don't know why
+          #
+          sed -i.orig "s|#configure=None|configure=--disable-maintainer-mode --enable-stuberrorlib --disable-shared --enable-shared=libgrp,stklib --disable-dependency-tracking|" setup.cfg
+
+          pip3 install -e .
+          pip3 install pytest
+          pip3 install pytest-xdist  # not really needed here
+          python3 -m pytest

--- a/sherpa/plot/tests/test_plot_notebook.py
+++ b/sherpa/plot/tests/test_plot_notebook.py
@@ -325,13 +325,22 @@ def test_regproj(old_numpy_printing, override_plot_backend):
 
     assert "<summary>RegionProjection (13)</summary>" in r
 
-    assert '<div class="dataname">parval0</div><div class="dataval">-0.5315772076542427</div>' in r
-    assert '<div class="dataname">parval1</div><div class="dataval">0.5854611101216837</div>' in r
+    # Issue #1372 shows that the numbers here can depend on the platform; as
+    # this test is not about whether the fit converged to the same solution
+    # the tests are very basic. An alternative would be to just place
+    # the values from the fit object into the strings, but then there is
+    # the problem that this test currently requires old_numpy_printing,
+    # so the results would not necessarily match.
+    #
+    assert '<div class="dataname">parval0</div><div class="dataval">-0.5' in r
+    assert '<div class="dataname">parval1</div><div class="dataval">0.5' in r
     assert '<div class="dataname">sigma</div><div class="dataval">(1, 2, 3)</div>' in r
 
-    assert '<div class="dataname">y</div><div class="dataval">[ 306.854444  282.795953  259.744431  237.699877  216.662291  196.631674\n' in r
+    # These values may depend on the platform so only very-limited check.
+    #
+    assert '<div class="dataname">y</div><div class="dataval">[ 30' in r
+    assert '<div class="dataname">levels</div><div class="dataval">[  3.6' in r
 
-    assert '<div class="dataname">levels</div><div class="dataval">[  3.606863   7.491188  13.140272]</div>' in r
     assert '<div class="dataname">min</div><div class="dataval">[-2, -1]</div>' in r
     assert '<div class="dataname">max</div><div class="dataval">[2, 2]</div>' in r
 

--- a/sherpa/plot/tests/test_plot_notebook.py
+++ b/sherpa/plot/tests/test_plot_notebook.py
@@ -1,5 +1,6 @@
 #
-# Copyright (C) 2020, 2021  Smithsonian Astrophysical Observatory
+# Copyright (C) 2020, 2021
+# Smithsonian Astrophysical Observatory
 #
 #
 #  This program is free software; you can redistribute it and/or modify
@@ -34,16 +35,20 @@ from sherpa.models.basic import Const1D, Gauss2D, Polynom1D
 from sherpa.stats import Chi2
 
 
+def plot_backend_is(name):
+    """Check we have the specified plot backend"""
+    return plot.backend.name == name
+
+
 def check_empty(r, summary, nsummary=0):
     """Is this an 'empty' response?"""
 
-    if plot.backend.name == 'pylab':
+    if plot_backend_is("pylab"):
         assert r is None
         return
 
     assert r is not None
-
-    assert '<summary>{} ({})</summary>'.format(summary, nsummary) in r
+    assert f"<summary>{summary} ({nsummary})</summary>" in r
 
 
 def check_full(r, summary, label, title, nsummary=0):
@@ -51,14 +56,14 @@ def check_full(r, summary, label, title, nsummary=0):
 
     assert r is not None
 
-    if plot.backend.name == 'pylab':
-        assert '<summary>{}</summary>'.format(summary) in r
-        assert '<svg ' in r
+    if plot_backend_is("pylab"):
+        assert f"<summary>{summary}</summary>" in r
+        assert "<svg " in r
         return
 
-    assert '<summary>{} ({})</summary>'.format(summary, nsummary) in r
-    assert '<div class="dataval">{}</div>'.format(label) in r
-    assert '<div class="dataval">{}</div>'.format(title) in r
+    assert f"<summary>{summary} ({nsummary})</summary>" in r
+    assert f'<div class="dataval">{label}</div>' in r
+    assert f'<div class="dataval">{title}</div>' in r
 
 
 def test_histogram(override_plot_backend):
@@ -205,13 +210,13 @@ def test_fit(override_plot_backend):
     # different to previous checks
     assert r is not None
 
-    if plot.backend.name == 'pylab':
-        assert '<summary>FitPlot</summary>' in r
-        assert '<svg ' in r
+    if plot_backend_is("pylab"):
+        assert "<summary>FitPlot</summary>" in r
+        assert "<svg " in r
         return
 
-    assert '<summary>DataPlot (' in r
-    assert '<summary>ModelPlot (' in r
+    assert "<summary>DataPlot (" in r
+    assert "<summary>ModelPlot (" in r
     assert '<div class="dataval">n n</div>' in r
     assert '<div class="dataval">Model</div>' in r
 
@@ -244,13 +249,13 @@ def test_fitcontour(override_plot_backend):
     # different to previous checks
     assert r is not None
 
-    if plot.backend.name == 'pylab':
-        assert '<summary>FitContour</summary>' in r
-        assert '<svg ' in r
+    if plot_backend_is("pylab"):
+        assert "<summary>FitContour</summary>" in r
+        assert "<svg " in r
         return
 
-    assert '<summary>DataContour (' in r
-    assert '<summary>ModelContour (' in r
+    assert "<summary>DataContour (" in r
+    assert "<summary>ModelContour (" in r
     assert '<div class="dataval">n n</div>' in r
     assert '<div class="dataval">Model</div>' in r
 
@@ -278,12 +283,12 @@ def test_intproj(old_numpy_printing, override_plot_backend):
     r = p._repr_html_()
     assert r is not None
 
-    if plot.backend.name == 'pylab':
-        assert '<summary>IntervalProjection</summary>' in r
-        assert '<svg ' in r
+    if plot_backend_is("pylab"):
+        assert "<summary>IntervalProjection</summary>" in r
+        assert "<svg " in r
         return
 
-    assert '<summary>IntervalProjection (8)</summary>' in r
+    assert "<summary>IntervalProjection (8)</summary>" in r
 
     assert '<div class="dataname">x</div><div class="dataval">[ 1.        1.555556  2.111111  2.666667  3.222222  3.777778  4.333333  4.888889\n  5.444444  6.      ]</div>' in r
     assert '<div class="dataname">nloop</div><div class="dataval">10</div>' in r
@@ -313,13 +318,12 @@ def test_regproj(old_numpy_printing, override_plot_backend):
     r = p._repr_html_()
     assert r is not None
 
-    if plot.backend.name == 'pylab':
-        assert '<summary>RegionProjection</summary>' in r
-        assert '<svg ' in r
+    if plot_backend_is("pylab"):
+        assert "<summary>RegionProjection</summary>" in r
+        assert "<svg " in r
         return
 
-    print(r)
-    assert '<summary>RegionProjection (13)</summary>' in r
+    assert "<summary>RegionProjection (13)</summary>" in r
 
     assert '<div class="dataname">parval0</div><div class="dataval">-0.5315772076542427</div>' in r
     assert '<div class="dataname">parval1</div><div class="dataval">0.5854611101216837</div>' in r

--- a/sherpa/tests/test_fit_unit.py
+++ b/sherpa/tests/test_fit_unit.py
@@ -91,7 +91,6 @@
 #
 
 import numpy as np
-from numpy.testing import assert_almost_equal
 
 import pytest
 
@@ -662,8 +661,8 @@ def test_fit_calc_stat_info_single(stat, usestat, usesys, expected, qval):
     # The __str__ and format methods use different formats when
     # displaying the numeric values. Unfortunately the format
     # method uses %g which - for the numbers here - can be less
-    # than 7 decimal places, so assert_almost_equal is not
-    # usable.
+    # than 7 decimal places, and initial testing with assert_almost_equal
+    # did not work so we used this approach instead.
     #
     def checkval_short(s, k, v):
         """Check that s == 'k = <v>'"""
@@ -2107,15 +2106,12 @@ def test_fit_multiple(stat, usestat, usesys, finalstat):
     assert fr.parnames[-1] == fr2.parnames[-1]
 
     if isinstance(statobj, Likelihood):
-        ndp = 0
+        ndp = 1
     else:
         ndp = 7
 
-    # pytest.approx does not have an equivalent of decimal;
-    # we could change this to an absolute tolerance
-    #
-    assert_almost_equal(fr2.parvals[-1], fr.parvals[-1],
-                        decimal=ndp)
+    rel = 10**(-ndp)
+    assert fr2.parvals[-1] == pytest.approx(fr.parvals[-1], rel=rel)
 
 
 @pytest.mark.parametrize("method,estmethod,usestat,usesys", [

--- a/sherpa/tests/test_fit_unit.py
+++ b/sherpa/tests/test_fit_unit.py
@@ -455,8 +455,8 @@ def test_fit_raises_error_on_bgsubtraction(stat):
     with pytest.raises(FitErr) as excinfo:
         fit.fit()
 
-    emsg = '{} statistics cannot be used with '.format(statobj.name) + \
-           'background subtracted data'
+    emsg = f"{statobj.name} statistics cannot be used with " + \
+           "background subtracted data"
     assert str(excinfo.value) == emsg
 
 
@@ -653,10 +653,10 @@ def test_fit_calc_stat_info_single(stat, usestat, usesys, expected, qval):
     # differences in precision/formatting can make this annoying.
     #
     def tostr_short(k, v):
-        return "{:9s} = {}".format(k, v)
+        return f"{k:9s} = {v}"
 
     def tostr_long(k, v):
-        return "{:21s} = {}".format(k, v)
+        return f"{k:21s} = {v}"
 
     # The __str__ and format methods use different formats when
     # displaying the numeric values. Unfortunately the format
@@ -676,7 +676,7 @@ def test_fit_calc_stat_info_single(stat, usestat, usesys, expected, qval):
 
         assert s.startswith(tostr_long(k, ''))
         sval = s.split(' = ')[1]
-        assert sval == "%g" % (v, )
+        assert sval == f"{v:g}"
 
     # Validate __str__
     #
@@ -1702,7 +1702,7 @@ def test_fit_str_single(stat):
                 ("stat", stat.__name__),
                 ("method", "LevMar"),
                 ("estmethod", "Covariance")]
-    expected = "\n".join(["{:9s} = {}".format(*e) for e in expected])
+    expected = "\n".join([f"{k:9s} = {v}" for (k,v) in expected])
 
     assert out == expected
 
@@ -1730,7 +1730,7 @@ def test_fit_str_multiple(stat):
                 ("stat", stat.__name__),
                 ("method", "LevMar"),
                 ("estmethod", "Covariance")]
-    expected = "\n".join(["{:9s} = {}".format(*e) for e in expected])
+    expected = "\n".join([f"{k:9s} = {v}" for (k,v) in expected])
 
     assert out == expected
 

--- a/sherpa/tests/test_fit_unit.py
+++ b/sherpa/tests/test_fit_unit.py
@@ -35,7 +35,8 @@
 # parameter values should also be adjusted to provide a better test of
 # the "q value" calculated for chi-square like statistics (at present
 # the value is often << 1e-7, which is the limit for
-# assert_almost_equal).
+# assert_almost_equal, although the tests are now being moved to
+# pytest.assert, which has similar behavior).
 #
 # Some of these tests may be superfluous (e.g. testing both the single
 # and multiple dataset handling for calc_chisqr with wstat), but given
@@ -547,7 +548,7 @@ def test_fit_calc_stat_single(stat, usestat, usesys, expected):
     # For now restrict the tests to using the default statistic values
     #
     fit = setup_stat_single(stat(), usestat, usesys)
-    assert_almost_equal(fit.calc_stat(), expected)
+    assert fit.calc_stat() == pytest.approx(expected)
 
 
 # Unfortunately the parameter choces mean that the qval values,
@@ -630,7 +631,7 @@ def test_fit_calc_stat_info_single(stat, usestat, usesys, expected, qval):
     assert ans.bkg_ids is None
     assert ans.numpoints == 5
     assert ans.dof == 1
-    assert_almost_equal(ans.statval, expected)
+    assert ans.statval == pytest.approx(expected)
 
     # The logic for the statistical name is a bit complicated.
     statname = statobj.name
@@ -645,9 +646,9 @@ def test_fit_calc_stat_info_single(stat, usestat, usesys, expected, qval):
         assert ans.rstat is None
 
     else:
-        assert_almost_equal(ans.qval, qval)
+        assert ans.qval == pytest.approx(qval)
         # as there's only 1 dof, statval == rstat
-        assert_almost_equal(ans.rstat, ans.statval)
+        assert ans.rstat == pytest.approx(ans.statval)
 
     # It is easiest just to do a direct string comparison, but
     # differences in precision/formatting can make this annoying.
@@ -669,7 +670,7 @@ def test_fit_calc_stat_info_single(stat, usestat, usesys, expected, qval):
 
         assert s.startswith(tostr_short(k, ''))
         sval = s.split(' = ')[1]
-        assert_almost_equal(float(sval), v)
+        assert float(sval) == pytest.approx(v)
 
     def checkval_long(s, k, v):
         """Check that s == 'k = <v>'"""
@@ -808,7 +809,7 @@ def test_fit_calc_chisqr_single(stat, usestat, usesys, expected):
     if expected is None:
         assert ans is None
     else:
-        assert_almost_equal(ans, expected)
+        assert ans == pytest.approx(expected)
 
 
 expected_multi_leastsq = 13837.0
@@ -895,12 +896,12 @@ def test_fit_calc_stat_multiple(stat, usestat, usesys, expected):
     statobj = stat()
     fit, fits = setup_stat_multiple(statobj, usestat, usesys, 1)
 
-    assert_almost_equal(fit.calc_stat(), expected)
+    assert fit.calc_stat() == pytest.approx(expected)
 
     # Test that the sum of the individual cases matches the expected
     # value.
     stats = [f.calc_stat() for f in fits]
-    assert_almost_equal(sum(stats), expected)
+    assert sum(stats) == pytest.approx(expected)
 
 
 # The choice of model parameter values does not create "good" fits,
@@ -1000,7 +1001,7 @@ def test_fit_calc_stat_info_multiple(stat, usestat, usesys, expected, qval):
     assert ans.bkg_ids is None
     assert ans.numpoints == nchannels
     assert ans.dof == ndof
-    assert_almost_equal(ans.statval, expected)
+    assert ans.statval == pytest.approx(expected)
 
     # The choice of ans.statname is interesting for chi-squared
     # cases, since it depends on whether usestat is True.
@@ -1015,8 +1016,8 @@ def test_fit_calc_stat_info_multiple(stat, usestat, usesys, expected, qval):
         assert ans.rstat is None
 
     else:
-        assert_almost_equal(ans.qval, qval)
-        assert_almost_equal(ans.rstat, ans.statval / ndof)
+        assert ans.qval == pytest.approx(qval, rel=1e-5)
+        assert ans.rstat == pytest.approx(ans.statval / ndof)
 
 
 chisqr_multi_leastsq = [4.0, 1.0, 9.0, 2500.0, 4900.0, 6400.0,
@@ -1147,9 +1148,9 @@ def test_fit_calc_chisqr_multiple(stat, usestat, usesys, expected):
         assert ans is None
         assert all([a is None for a in allans])
     else:
-        assert_almost_equal(ans, expected)
+        assert ans == pytest.approx(expected, rel=1e-5)
         combined = np.concatenate(allans)
-        assert_almost_equal(combined, expected)
+        assert combined == pytest.approx(expected, rel=1e-5)
 
 
 wstat_single_scalar_stat = 18.907726418126835
@@ -1224,7 +1225,7 @@ def test_fit_calc_stat_wstat_single(scalar, usestat, usesys,
     """
 
     fit = setup_pha_single(scalar, usestat, usesys, flo, fhi)
-    assert_almost_equal(fit.calc_stat(), expected)
+    assert fit.calc_stat() == pytest.approx(expected)
 
 
 # The qval values were calculated using changeset
@@ -1307,7 +1308,7 @@ def test_fit_calc_stat_info_wstat_single(scalar, usestat, usesys,
     # mdl.c0 is the only thawed parameter
     dof = nbin - 1
 
-    assert_almost_equal(fit.calc_stat(), expected)
+    assert fit.calc_stat() == pytest.approx(expected)
     assert isinstance(ans, StatInfoResults)
     assert ans.name == ""
     assert ans.statname == "wstat"
@@ -1315,10 +1316,10 @@ def test_fit_calc_stat_info_wstat_single(scalar, usestat, usesys,
     assert ans.bkg_ids is None
     assert ans.numpoints == nbin
     assert ans.dof == dof
-    assert_almost_equal(ans.statval, expected)
+    assert ans.statval == pytest.approx(expected)
 
-    assert_almost_equal(ans.qval, qval)
-    assert_almost_equal(ans.rstat, expected / dof)
+    assert ans.qval == pytest.approx(qval)
+    assert ans.rstat == pytest.approx(expected / dof)
 
 
 # Do not really need to go through all these options, but it's easy
@@ -1434,7 +1435,7 @@ def test_fit_calc_stat_wstat_grouped_single(flo, fhi, expected):
 
     # For now restrict to the default statistic values
     fit = Fit(src, mdl, stat=WStat())
-    assert_almost_equal(fit.calc_stat(), expected)
+    assert fit.calc_stat() == pytest.approx(expected)
 
 
 wstat_multi_all = 28.114709948
@@ -1471,12 +1472,12 @@ def test_fit_calc_stat_wstat_multiple(flo, fhi, expected):
     """
 
     fit, fits = setup_pha_multiple(flo, fhi)
-    assert_almost_equal(fit.calc_stat(), expected)
+    assert fit.calc_stat() == pytest.approx(expected)
 
     # Test that the sum of the individual cases matches the expected
     # value.
     stats = [f.calc_stat() for f in fits]
-    assert_almost_equal(sum(stats), expected)
+    assert sum(stats) == pytest.approx(expected)
 
 
 # Remember, the filter is only applied to the background for the third
@@ -1522,7 +1523,7 @@ def test_fit_calc_stat_info_wstat_multiple(flo, fhi, nbin, expected, qval):
 
     dof = nbin - 5
 
-    assert_almost_equal(fit.calc_stat(), expected)
+    assert fit.calc_stat() == pytest.approx(expected)
     assert isinstance(ans, StatInfoResults)
     assert ans.name == ""
     assert ans.statname == "wstat"
@@ -1530,10 +1531,10 @@ def test_fit_calc_stat_info_wstat_multiple(flo, fhi, nbin, expected, qval):
     assert ans.bkg_ids is None
     assert ans.numpoints == nbin
     assert ans.dof == dof
-    assert_almost_equal(ans.statval, expected)
+    assert ans.statval == pytest.approx(expected)
 
-    assert_almost_equal(ans.qval, qval)
-    assert_almost_equal(ans.rstat, expected / dof)
+    assert ans.qval == pytest.approx(qval)
+    assert ans.rstat == pytest.approx(expected / dof)
 
 
 @pytest.mark.parametrize("flo,fhi,expected", [
@@ -1670,13 +1671,13 @@ def test_fit_calc_stat_error_no_cache():
     def ans(delta, dy):
         return (delta / dy) ** 2
 
-    assert_almost_equal(f.calc_stat(), ans(1.0, 1.3))
+    assert f.calc_stat() == pytest.approx(ans(1.0, 1.3))
 
     d.y = np.asarray([3.1, 4.1])
     d.staterror = np.asarray([1.1, 1.5])
     m.c0 = 4.1
 
-    assert_almost_equal(f.calc_stat(), ans(1.0, 1.1))
+    assert f.calc_stat() == pytest.approx(ans(1.0, 1.1))
 
 
 @pytest.mark.parametrize("stat", [
@@ -1879,7 +1880,7 @@ def test_fit_single(stat, usestat, usesys, finalstat):
     assert fit.method.name == 'levmar'
     fr = fit.fit()
     assert fr.succeeded
-    assert_almost_equal(fr.statval, finalstat)
+    assert fr.statval == pytest.approx(finalstat)
 
 
 @pytest.mark.parametrize("stat,usestat,usesys,finalstat", [
@@ -1909,7 +1910,7 @@ def test_fit_single_nm(stat, usestat, usesys, finalstat):
     fit.method = NelderMead()
     fr = fit.fit()
     assert fr.succeeded
-    assert_almost_equal(fr.statval, finalstat)
+    assert fr.statval == pytest.approx(finalstat)
 
 
 # Since the background is being ignored in this fit (except for
@@ -1999,7 +2000,7 @@ def test_fit_single_pha(stat, scalar, usestat, usesys, filtflag, finalstat):
     assert fr.succeeded
     assert fr.numpoints == numpoints
     assert fr.dof == (numpoints - 1)
-    assert_almost_equal(fr.statval, finalstat)
+    assert fr.statval == pytest.approx(finalstat)
 
 
 # Calculated using LevMar
@@ -2081,12 +2082,12 @@ def test_fit_multiple(stat, usestat, usesys, finalstat):
     fit, _ = setup_stat_multiple(statobj, usestat, usesys, 1)
     fr = fit.fit()
     assert fr.succeeded
-    assert_almost_equal(fr.statval, finalstat)
+    assert fr.statval == pytest.approx(finalstat)
 
     fit, _ = setup_stat_multiple(statobj, usestat, usesys, 3)
     fr = fit.fit()
     assert fr.succeeded
-    assert_almost_equal(fr.statval, finalstat)
+    assert fr.statval == pytest.approx(finalstat)
 
     # As the datasets and models are independent, the fit
     # to just the third dataset should return the same parameter
@@ -2110,6 +2111,9 @@ def test_fit_multiple(stat, usestat, usesys, finalstat):
     else:
         ndp = 7
 
+    # pytest.approx does not have an equivalent of decimal;
+    # we could change this to an absolute tolerance
+    #
     assert_almost_equal(fr2.parvals[-1], fr.parvals[-1],
                         decimal=ndp)
 
@@ -2452,11 +2456,11 @@ def test_fit_iterfit_single_sigmarej_chi2(stat):
     expected_mask = [True, True, False, False, False, False, False]
     assert np.all(fit.data.mask == expected_mask)
 
-    assert_almost_equal(fr.statval, 0.0)
+    assert fr.statval == pytest.approx(0.0)
 
     mdl = fit.model
-    assert_almost_equal(mdl.c0.val, 8.722)
-    assert_almost_equal(mdl.c1.val, 2.458)
+    assert mdl.c0.val == pytest.approx(8.722)
+    assert mdl.c1.val == pytest.approx(2.458)
 
     assert fr.numpoints == 2
     assert fr.dof == 0
@@ -2483,11 +2487,11 @@ def test_fit_iterfit_single_sigmarej_chi2gehrels():
     expected_mask = [True, True, True, True, False, True, True]
     assert np.all(fit.data.mask == expected_mask)
 
-    assert_almost_equal(fr.statval, 0.1914790757)
+    assert fr.statval == pytest.approx(0.1914790757)
 
     mdl = fit.model
-    assert_almost_equal(mdl.c0.val, 9.69168196604)
-    assert_almost_equal(mdl.c1.val, 2.00600360315)
+    assert mdl.c0.val == pytest.approx(9.69168196604)
+    assert mdl.c1.val == pytest.approx(2.00600360315)
 
     assert fr.numpoints == 6
     assert fr.dof == 4
@@ -2517,11 +2521,11 @@ def test_fit_iterfit_single_sigmarej_ignore_chi2gehrels():
     expected_mask = [True, False, True, True, False, True, True]
     assert np.all(fit.data.mask == expected_mask)
 
-    assert_almost_equal(fr.statval, 0.1245627587)
+    assert fr.statval == pytest.approx(0.1245627587)
 
     mdl = fit.model
-    assert_almost_equal(mdl.c0.val, 9.25537857670)
-    assert_almost_equal(mdl.c1.val, 2.01845980545)
+    assert mdl.c0.val == pytest.approx(9.25537857670)
+    assert mdl.c1.val == pytest.approx(2.01845980545)
 
     assert fr.numpoints == 5
     assert fr.dof == 3

--- a/sherpa/tests/test_fit_unit.py
+++ b/sherpa/tests/test_fit_unit.py
@@ -1879,7 +1879,7 @@ def test_fit_single(stat, usestat, usesys, finalstat):
     assert fit.method.name == 'levmar'
     fr = fit.fit()
     assert fr.succeeded
-    assert fr.statval == pytest.approx(finalstat)
+    assert fr.statval == pytest.approx(finalstat, rel=7e-5)
 
 
 @pytest.mark.parametrize("stat,usestat,usesys,finalstat", [

